woocommerce/woocommerce
# New filter to set a custom precision value for the WC calculations

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR adds a new filter to set a custom precision value for the WC calculations: `wc_get_rounding_precision`.



### How to test the changes in this Pull Request:

1. Set the option "Yes, I will enter prices inclusive of tax" under WooCommerce > Settings > Tax > Prices entered with tax:
   
   ![image](https://github.com/woocommerce/woocommerce/assets/38109855/01a23986-702c-4712-84c1-7f352c97dc94)
2. Insert a row and enter 21% as a standard tax rate (it will help to generate a price with decimals and it is also an actual tax rate in several EU countries):
   
   ![image](https://github.com/woocommerce/woocommerce/assets/38109855/adc57de4-4f8a-4cb7-9a6d-461ebe8c9164)

3. Place a new order, selecting any product.
4. Edit the item within the order details screen, and check the decimal numbers:
   
   ![image](https://github.com/woocommerce/woocommerce/assets/38109855/341d481b-16a6-4b4a-b44b-71e73d4f0ab3)
5. Activate the following code snippet in the testing site:

```php
/**
 * WooCommerce:
 * Customize the precision value for WC calculations
 */
add_filter( 'wc_get_rounding_precision', function( $precision ) {
	return 2;
}, 10, 1 );
```
6. Place a second order, selecting the same product chosen for the first order.
7. Edit the item within the order details screen for the second order, and check the decimal numbers (they should be only 2 this time):
   
    ![image](https://github.com/woocommerce/woocommerce/assets/38109855/56c88d9d-d4eb-4337-85cc-45a0505dbc99)



### Changelog entry

New filter to set a custom precision value for the WC calculations: `wc_get_rounding_precision`




<details>

#### Significance




#### Type



</details>
